torch/onnx/_internal/fx/fx_exporter.py

```diff
@@ -129,6 +129,58 @@ def apply(
         return tuple(arg for arg in model_args if arg is not None), {}
 
 
+class RemoveConstantInputStep:
+    """Remove the constant input arguments.
+
+    Dynamo does not support constant input arguments (https://github.com/pytorch/pytorch/issues/99534).
+    But it does put the input into graph with an empty node, and consumed by no ones.
+    For example,
+
+        def func(x, b=1.0):
+            y = x + b
+            z = y.relu()
+            return (y, z)
+
+        x = torch.randn(1, 1, 2, dtype=torch.float32)
+        gm_fun, _ = dynamo.export(func, x, b=8.0, aten_graph=True, tracing_mode="real")
+
+        # class GraphModule(torch.nn.Module):
+        #     def forward(self, x, b):
+        #         arg0: f32[1, 1, 2], arg1, = fx_pytree.tree_flatten_spec(([x, b], {}), self._in_spec)
+        #         # File: path/to/pytorch/test_constant_input.py:5, code: y = x + b
+        #         add_tensor: f32[1, 1, 2] = torch.ops.aten.add.Tensor(arg0, 8.0);  arg0 = None
+
+        #         # File: path/to/pytorch/test_constant_input.py:6, code: z = y.relu()
+        #         relu_default: f32[1, 1, 2] = torch.ops.aten.relu.default(add_tensor)
+        #         return pytree.tree_unflatten([add_tensor, relu_default], self._out_spec)
+
+    ONNX graph deletes None input leading to a mismatched number of input with PyTorch.
```

Comment:
There is a bit of nuance here. Why does that input end up as `None input`? This is different to real `None input` cases, where `None` is passed in.

Because the alternative is, if this kind of input is perceived as its original type (float number) by ONNX graph builder, and are actually supported. The issue will now only be that we got an unused scalar input in the graph. In that case we won't need the input adapter.

Reply:
The story is that fx gives us `torch.fx.Node` with nothing inside it. In torch perspective, we say it's different from None input, but in ONNX perspective, it's None input just like real None input. I can modify this to more close to torch perspective.

Reply:
Rephrased.